opcodes.h

```diff
@@ -0,0 +1,286 @@
+  _fmt(OP_MIX_APPEND,                     N_("append a remailer to the chain")) \
+  _fmt(OP_MIX_INSERT,                     N_("insert a remailer into the chain")) \
+  _fmt(OP_MIX_DELETE,                     N_("delete a remailer from the chain")) \
+  _fmt(OP_MIX_CHAIN_PREV,                 N_("select the previous element of the chain")) \
+  _fmt(OP_MIX_CHAIN_NEXT,                 N_("select the next element of the chain")) \
+  _fmt(OP_COMPOSE_MIX,                    N_("send the message through a mixmaster remailer chain"))
+#else
+#define OPS_MIX(_)
+#endif
+
+#ifdef USE_NOTMUCH
+#define OPS_NOTMUCH(_fmt) \
+  _fmt(OP_MAIN_CHANGE_VFOLDER,            N_("open a different virtual folder")) \
+  _fmt(OP_MAIN_VFOLDER_FROM_QUERY,        N_("generate virtual folder from query")) \
+  _fmt(OP_MAIN_WINDOWED_VFOLDER_FORWARD,  N_("shifts virtual folder time window forwards")) \
+  _fmt(OP_MAIN_WINDOWED_VFOLDER_BACKWARD, N_("shifts virtual folder time window backwards")) \
+  _fmt(OP_MAIN_MODIFY_LABELS,             N_("modify (notmuch) tags")) \
+  _fmt(OP_MAIN_MODIFY_LABELS_THEN_HIDE,   N_("modify labels and then hide message")) \
+  _fmt(OP_MAIN_ENTIRE_THREAD,             N_("read entire thread of the current message"))
+#else
+#define OPS_NOTMUCH(_)
+#endif
+
+#define OPS_PGP(_fmt) \
+  _fmt(OP_COMPOSE_ATTACH_KEY,             N_("attach a PGP public key")) \
+  _fmt(OP_COMPOSE_PGP_MENU,               N_("show PGP options")) \
+  _fmt(OP_MAIL_KEY,                       N_("mail a PGP public key")) \
+  _fmt(OP_VERIFY_KEY,                     N_("verify a PGP public key")) \
+  _fmt(OP_VIEW_ID,                        N_("view the key's user id")) \
+  _fmt(OP_CHECK_TRADITIONAL,              N_("check for classic PGP")) \
+
+#define OPS_SIDEBAR(_fmt) \
+  _fmt(OP_SIDEBAR_NEXT,                   N_("move the highlight to next mailbox")) \
+  _fmt(OP_SIDEBAR_NEXT_NEW,               N_("move the highlight to next mailbox with new mail")) \
+  _fmt(OP_SIDEBAR_OPEN,                   N_("open highlighted mailbox")) \
+  _fmt(OP_SIDEBAR_PAGE_DOWN,              N_("scroll the sidebar down 1 page")) \
+  _fmt(OP_SIDEBAR_PAGE_UP,                N_("scroll the sidebar up 1 page")) \
+  _fmt(OP_SIDEBAR_PREV,                   N_("move the highlight to previous mailbox")) \
+  _fmt(OP_SIDEBAR_PREV_NEW,               N_("move the highlight to previous mailbox with new mail")) \
+  _fmt(OP_SIDEBAR_TOGGLE_VIRTUAL,         N_("toggle between mailboxes and virtual mailboxes")) \
+  _fmt(OP_SIDEBAR_TOGGLE_VISIBLE,         N_("make the sidebar (in)visible")) \
+
+#define OPS_SMIME(_fmt) \
+  _fmt(OP_COMPOSE_SMIME_MENU,             N_("show S/MIME options")) \
+
+
+#define OPS(_fmt) \
+  OPS_CORE(_fmt) \
+  OPS_SIDEBAR(_fmt) \
+  OPS_MIX(_fmt) \
+  OPS_NOTMUCH(_fmt) \
+  OPS_PGP(_fmt) \
+  OPS_SMIME(_fmt) \
+  OPS_CRYPT(_fmt) \
+
+enum mutt_ops {
+#define DEFINE_OPS(opcode, help_string) opcode,
+  OPS(DEFINE_OPS)
+  OP_MAX,
```

Comment:
When doing thing kind of expnsion, I usually `#undef` the temporary macro just after use. I find it makes it clearer what it's his scope. Here, I'd do that just after the expansion of `OPS`, or if it looks nicer just before the `enum` closing brace.